frontend/src/App.js

```diff
@@ -51,16 +76,27 @@ const uiSchemaForSchema = (schema) => {
 }
 
 const ConfiguratorForm = ({ schema, formData }) => {
+  const [isLoading, setLoading] = useState(false);
+  const [alertVariant, setAlertVariant] = useState("bg-dark text-white");
+  const [alertText, setAlertText] = useState("Click the submit button to save your changes");
+  const [alertShow, setAlertVisibility] = useState(true);
+
   return (
     <Form
       schema={schema}
       uiSchema={uiSchemaForSchema(schema)}
       onChange={console.log}
-      onSubmit={updateConfig}
+      onSubmit={e => updateConfig(e, setLoading, setAlertVariant, setAlertText, setAlertVisibility)}
       formData={formData}
       widgets={customWidgets}
-      onError={console.log}
-    />
+      onError={console.log}>
+      <Toast onClose={() => setAlertVisibility(false)} show={alertShow} delay={2000} autohide>
+        <Toast.Body className={alertVariant}>{alertText}</Toast.Body>
+      </Toast>
+      <Button type="submit" variant="primary" disabled={isLoading}>
+        {isLoading ? 'Loading…' : 'Submit'}
+      </Button>
```

Comment:
What about putting the toast after the button?
From the gif you posted, having the toast above is causing the button to be "dancing" and users usually expect buttons to stay in their place.
WDYT?

Reply:
I experimented a bit with placing the toast in different places :thinking: 

* This is how it looks with the toast at the bottom:
![toast-at-the-bottom](https://user-images.githubusercontent.com/7579677/115418959-9b0eba00-a202-11eb-8dd3-465c522ba9bc.gif)

* This is how it looks with the toast in the right side:
![toat-at-right](https://user-images.githubusercontent.com/7579677/115419076-b37ed480-a202-11eb-8521-90bf2b5f9e81.gif)

The submit button seems it's not well aligned with the rest of the form though.

What do you think?

Reply:
Both are OK, but if you go with the first option, the toast should be separated from the Submit button and if you go with the second option, you should try to fix that subtle change in the button height once the toast appears, IMHO.